server/src/features/orders/controller.ts

```diff
@@ -125,3 +122,104 @@ export class ProductOrdersController extends Controller {
     return new OrderService().deleteProductOrderById(productOrderId);
   }
 }
+
+@Route("create-checkout-session")
+@Tags("Orders")
+export class PaymentController extends Controller {
+  @Security("jwt")
+  @Post()
+  public async createCheckoutSession(
+    @Request() req: ExRequest,
+    @Body() requestBody: CheckoutInfo
+  ): Promise<void> {
+    const line_items = requestBody.order.map((item: any) => {
+      return {
+        price_data: {
+          currency: "eur",
+          product_data: {
+            name: item.productId,
+          },
+          unit_amount: item.price * 100,
+        },
+        quantity: item.quantity,
+      };
+    });
+    try {
+      const session = await stripe.checkout.sessions.create({
+        mode: "payment",
+        customer_email: requestBody.userEmail,
+        line_items,
+        success_url: `http://localhost:3000/successfull`,
+        cancel_url: `http://localhost:3000/cancelled`,
+      });
+      req.res?.json({ url: session.url });
+    } catch (e: any) {
+      req.res?.status(500).json({ error: e.message });
+    }
+  }
+}
+const prisma = new PrismaClient();
+const endpointSecret = ENDPOINT_SECRET;
+
+@Route("/webhook")
+@Tags("Orders")
+export class WebhookController extends Controller {
+  @Post()
+public async createWebhook(
+  @Request() req: ExRequest,
+): Promise<void>{
+  let event = req.body;
+    // Only verify the event if you have an endpoint secret defined.
+    // Otherwise use the basic event deserialized with JSON.parse
+    if (endpointSecret) {
+      // Get the signature sent by Stripe
+      const signature = req.headers["stripe-signature"];
+      try {
+        event = stripe.webhooks.constructEvent(
+          req.body.text(),
+          signature!,
+          endpointSecret
+        );
+      } catch (err: any) {
+        console.log(`⚠️  Webhook signature verification failed.`, err.message);
+      }
+    }
+    // Handle the event
+    switch (event.type) {
+      case "checkout.session.completed":
+        const checkoutSessionCompleted = event.data.object;
+        //Then define and call a function to handle the event checkout.session.completed
+        const user = await prisma.user.findUnique({
+          where: {
+            email: event.data.object.customer_details.email,
+          },
+        });
+
+        const prismaOrder = await prisma.order.findFirst({
+          where: {
+            userId: user!.id,
+          },
+          orderBy: {
+            id: "desc",
+          },
+        });
+
+        await prisma.order.update({
+          where: {
+            id: prismaOrder!.id,
+          },
+          data: {
+            status: "Payment Received",
+            amount: event.data.object.amount_total / 100,
+          },
+        });
+        console.log(checkoutSessionCompleted);
+        break;
+      default:
+        console.log(`Unhandled event type ${event.type}`);
+    }
+    req.res?.send();
+}
+}
+
```

Comment:
1. Should be in a stripe prefix (since it's stripe logic)
2. Lots of code that has nothing to do with controllers, should in services, models and helpers